### DISCUSSION TITLE
tar_read inside test_that

### DISCUSSION THREAD

**Discussion by CorradoLanera** [ropensci/targets#726] [Help] [answered]
I am developing a project using `{targets}`, testing custom function w/ `{testthat}`. When I have tested all the functions helpful in creating a target, I would like to use that object in a further test, avoiding creating it every time.

Is it possible to do the following works? Or what is the suggested approach?

``` r
# tests/testthat/test-preprocess_raw.R

test_that("I am able to preprocess imported (targets) data", {
  # setup
  raw <- tar_load(raw_imported) 

  # evaluation
  prep <- preprocess(raw)

  # expectation
  expect_s3_class(prep, "data.frame")
})
```

Current result is

``` r
Error (test-preprocess_raw.R:3:3): multiplication works
<tar_condition_validate/tar_condition_targets/rlang_error/error/condition>
Error: missing files: _targets/meta/meta
Backtrace:
 1. targets::tar_read(raw_imported) test-preprocess_raw.R:3:2
 3. targets::tar_meta(store = store)
 4. targets::tar_assert_path(path_meta(path_store = store))
 5. targets::tar_throw_validate(...)
 ```

I suppose that is because tests are running in a different folder with respect to the project one, so `_targets` is not found.

**Comment by wlandau** [comment, marked as answer]
Yeah, that's because the test's working directory is the `testthat` directory, not the root of the project where `_targets/` is. In the test, you could call `tar_load(raw_imported, store = "../../_targets")`. Alternatively, you could put a `_targets.yaml` in the `testthat` directory that points to the correct data store: http://books.ropensci.org/targets/projects.html
